vsintegration/src/FSharp.Editor/Navigation/FindUsagesService.fs

```diff
@@ -23,22 +23,23 @@ type internal FSharpFindUsagesService
         projectInfoManager: ProjectInfoManager
     ) =
     
-    // File can be included in more than one project, hence single `range` may results with multiple `Document`s.
-    let rangeToDocumentSpans (solution: Solution, range: range, cancellationToken: CancellationToken) =
+    // File can be included in more than one project.
+    let rangeToDocumentSpan (solution: Solution, range: range, cancellationToken: CancellationToken) =
         async {
             if range.Start = range.End then return []
             else 
                 let! spans =
                     solution.GetDocumentIdsWithFilePath(range.FileName)
                     |> Seq.map (fun documentId ->
-                        async {
+                        asyncMaybe {
                             let doc = solution.GetDocument(documentId)
                             let! sourceText = doc.GetTextAsync(cancellationToken)
                             match CommonRoslynHelpers.TryFSharpRangeToTextSpan(sourceText, range) with
                             | Some span ->
                                 let span = CommonHelpers.fixupSpan(sourceText, span)
-                                return Some (DocumentSpan(doc, span))
-                            | None -> return None
+                                return DocumentSpan(doc, span)
```

Comment:
Basically, this code switched from `async` to `asyncMaybe` which remove the need to spell `Some`, and also remove some nesting (not in this diff, but you'll see it in other places) when obtaining an `option` and continuing on the `Some` branch is made without need to pattern match.

Notice that the None which is returned is using `return!` instead of `return`.

I know it looks tricky, but in the grand scheme it reduces a good amount of accidental complexity dealing with `Async<T>` and `Async<T option>`, this can have significant effect on the code handling several such calls.